src/CarbonAware.WebApi/test/integrationTests/CarbonAwareControllerTests.cs

```diff
@@ -20,18 +20,73 @@ public class CarbonAwareControllerTests : IntegrationTestingBase
     private readonly string healthURI = "/health";
     private readonly string fakeURI = "/fake-endpoint";
     private readonly string bestLocationsURI = "/emissions/bylocations/best";
+    private readonly string bylocationsURI = "/emissions/bylocations";
+    private readonly string bylocationURI = "/emissions/bylocation";
     private readonly string currentForecastURI = "/emissions/forecasts/current";
     private readonly string batchForecastURI = "/emissions/forecasts/batch";
     private readonly string averageCarbonIntensityURI = "/emissions/average-carbon-intensity";
     private readonly string batchAverageCarbonIntensityURI = "/emissions/average-carbon-intensity/batch";
 
+    /// <summary>
+    /// Exected responses from mocks depending on DataSourceType, TestCase and test method.
+    /// </summary>
+    private readonly Dictionary<DataSourceType, Dictionary<string, string>> expectedResults =
+        new Dictionary<DataSourceType, Dictionary<string, string>>()
+        {
+            {
+                DataSourceType.JSON, new Dictionary<string, string>()
```

Comment:
I'm wondering about this data was generated. Is there any better way to have this content in a more sustainable way? It feels very error prone.

Reply:
@juzuluag switched to snapshot testing. cc: @JenMadiedo @Kpopovs1